themes/V3/5ePHB/snippets/tableOfContents.gen.js

```diff
@@ -1,41 +1,65 @@
 const _ = require('lodash');
 const dedent = require('dedent-tabs').default;
 
-const getTOC = (pages)=>{
+const getTOC = ()=>{
 
-	const recursiveAdd = (title, page, targetDepth, child, curDepth=0)=>{
+	const iframe = document.getElementById('BrewRenderer');
+	const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
+	const headings = iframeDocument.querySelectorAll('h1, h2, h3, h4, h5, h6');
+	const headerDepth = ['H1', 'H2', 'H3', 'H4', 'H5', 'H6'];
+
+	const res = [];
+
+	const pageMap = [];
+
+	const walkPages = ()=>{
+		let current = 0;
+		let skip = 0;
+		let reset = 0;
+		const pages = iframeDocument.querySelectorAll('.page');
+		_.each(pages, (page)=>{
+			current++;
+			if(page.querySelectorAll('.skipCounting').length > 0) {
+				skip += 1;
+			} else if(page.querySelectorAll('.resetCounting').length > 0) {
+				reset = current - 1;
+				skip = 0;
+			}
+			pageMap[current] = current - reset - skip;
+		});
+	};
+
+	const recursiveAdd = (title, page, anchor, targetDepth, child, curDepth=0)=>{
 		if(curDepth > 5) return; // Something went wrong.
 		if(curDepth == targetDepth) {
 			child.push({
 				title    : title,
 				page     : page,
+				anchor   : anchor,
 				children : []
 			});
 		} else {
 			if(child.length == 0) {
 				child.push({
 					title    : null,
 					page     : page,
+					anchor   : anchor,
 					children : []
 				});
 			}
-			recursiveAdd(title, page, targetDepth, _.last(child).children, curDepth+1,);
+			recursiveAdd(title, page, anchor, targetDepth, _.last(child).children, curDepth+1,);
 		}
 	};
 
-	const res = [];
-
-	const iframe = document.getElementById('BrewRenderer');
-	const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
-	const headings = iframeDocument.querySelectorAll('h1, h2, h3, h4, h5, h6');
-	const headerDepth = ['H1', 'H2', 'H3', 'H4', 'H5', 'H6'];
+	walkPages();
 
 	_.each(headings, (heading)=>{
+		const pageAnchor = heading.closest('.page').id;
 		const onPage = parseInt(heading.closest('.page').id?.replace(/^p/, ''));
```

Comment:
So `pageAnchor` is always identical to `onPage` except a `p` is in front of the number, right?.

I wonder if we can make the logic easier to parse if we just use `onPage` throughout (they are the same number), and then just add the `p` back at the end when emitting the markdown.

Reply:
It's not. But it's close enough I can fake it